orchestrator/careplancontributor/service.go

```diff
@@ -259,7 +244,43 @@ func (s Service) withSessionOrBearerToken(next func(response http.ResponseWriter
 	}
 }
 
-func (s Service) handleNotification(_ any) error {
-	log.Info().Msg("TODO: Handle received notification")
+func (s Service) handleNotification(resource any) error {
+	notification := resource.(*coolfhir.SubscriptionNotification)
+	if notification == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "failed to cast resource to notification",
+			StatusCode: http.StatusInternalServerError,
+		}
+	}
+
+	focusReference, err := notification.GetFocus()
+	if err != nil {
+		return err
+	}
+	if focusReference == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "Notification focus now found",
+			StatusCode: http.StatusInternalServerError,
+		}
+	}
+
+	switch *focusReference.Type {
+	case "Task":
+		// Get task
+		var task fhir.Task
+		err = s.carePlanServiceClient.Read(*focusReference.Reference, &task)
+		if err != nil {
+			return err
+		}
+
+		// TODO: How to differentiate between create and update? (Currently we only use Create in CPS. There is code for Update but nothing calls it)
```

Comment:
I am not sure if something is built into the notification solution we use, we need to check that first. If not:

Every task has a version in the `Task.meta.versionId`. Mostly, this is an incremental integer, but FHIR clearly states that the code must not assume this to be an incremental integer.

> Changes each time the content of the resource changes. Can be referenced in a [resource reference](https://www.hl7.org/fhir/r4/references.html#Resource). Can be used to ensure that updates are based on the latest version of the resource.
The version can be globally unique, or scoped by the Logical Id of the resource. Version identifiers are generally either a serially incrementing id scoped by the logical id, or a uuid, though neither of these approaches is required. There is no fixed order for version ids - clients cannot assume that a versionId that comes after another one either numerically or alphabetically represents a later version. The same versionId can never be used for more than one version of the same resource.
On the RESTful API: On receiving a [write operation](https://www.hl7.org/fhir/r4/http.html#write), the server SHALL update this item to the current value, or remove it. Note that servers SHOULD support versions, but some are unable to

So, if we cannot rely on the versionId, a safer (but less performant) option is to:
```
GET [baseURL]/Task/[id]/_history?_summary=count
```
This will provide the count of versions for a given Task.

Reply:
I am hesitant to add this change in this PR considering the task was to simply move the functionality and I would want to modify it as little as needed. That being said, if it is required for feature parity (considering this is now in CPC instead of CPS and is now an async process), then it would need to be implemented.

I am not fully aware of the impact so I will bring it up during the team discussions on Monday

Reply:
For sure, it was more of an informative comment answering the TODO, indeed out of scope for this PR!